### PR TITLE
doc: add a doc about how to configure  supernode,dfget,dfdaemon

### PR DESCRIPTION
**Pull request by yunfeiyanggzq**
Signed-off-by: yunfeiyangbuaa <yunfeiyang@buaa.edu.cn>

### Ⅰ. Describe what this PR did
reference #802

### CODE REVIEW

**Comment by yunfeiyanggzq** [comment]
@Starnop

File: docs/config/supernode_properties.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Customize supernode properties
+
+This topic explains how to customize the dragonfly supernode startup parameters.
+
+## Parameter instructions
+
+### The parameters we can configure in supernode is as follows
```
[the remark this answers is not in the document]
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: you are right

File: docs/config/supernode_properties.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+| downloadPort | 8001 | downloadPort is the port for download files from supernode |
+| homeDir | /home/admin/supernode | homeDir is the working directory of supernode |
+| advertiseIP | localhost | the supernode ip is the ip we advertise to other peers in the p2p-network |
+| schedulerCorePoolSize | 10 | pool size is the core pool size of ScheduledExecutorService |
+| downloadPath | /home/admin/supernode/repo/download | DownloadPath specifies the path where to store downloaded files from source address |
+| peerUpLimit | 5 | upload limit for a peer to serve download tasks |
+| peerDownLimit | 4 |the task upload limit of a peer when dfget starts to play a role of peer |
+| eliminationLimit | 5 | if a dfget fails to provide service for other peers up to eliminationLimit, it will be isolated |
+| failureCountLimit | 5 | when dfget client fails to finish distribution task up to failureCountLimit, supernode will add it to blacklist|
+| linkLimit | 20480GB | LinkLimit is set for supernode to limit every piece download network speed |
```
[the remark this answers is not in the document]
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: I will check it

File: docs/config/supernode_properties.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+### The parameters we can configure in supernode is as follows
+
+The following startup parameters are supported for `supernode`
+
+| Parameter | Default | Description |
+| ------------- | ------------- | ------------- |
+| listenPort | 8002 | listenPort is the port that supernode server listens on |
+| downloadPort | 8001 | downloadPort is the port for download files from supernode |
+| homeDir | /home/admin/supernode | homeDir is the working directory of supernode |
+| advertiseIP | localhost | the supernode ip is the ip we advertise to other peers in the p2p-network |
```
Comment (Starnop) [inline docs/config/supernode_properties.md]: the first non-loop address

File: docs/config/supernode_properties.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+| schedulerCorePoolSize | 10 | pool size is the core pool size of ScheduledExecutorService |
```
Comment (Starnop) [inline docs/config/supernode_properties.md]: In fact, `schedulerCorePoolSize` is unused for now. And make a record here.

File: docs/config/supernode_properties.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Starnop) [reply docs/config/supernode_properties.md]: In fact, there is an error here. 

```
LinkLimit:               DefaultLinkLimit * rate.MB,
SystemReservedBandwidth: DefaultSystemReservedBandwidth * rate.MB,
MaxBandwidth:            DefaultMaxBandwidth * rate.MB,
```
And the default values are as follows:

```
DefaultLinkLimit = 20 * rate.MB
DefaultSystemReservedBandwidth = 20 * rate.MB
DefaultMaxBandwidth = 200 * rate.MB
```
They are both multiply by rate.MB. So could you please fix that? @yunfeiyanggzq THX

File: docs/config/supernode_properties.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+| failAccessInterval | 3m0s | fail access interval is the interval time after failed to access the URL |
+| gcInitialDelay | 6s | gc initial delay is the delay time from the start to the first GC execution |
+| gcMetaInterval | 2m0s | gc meta interval is the interval time to execute the GC meta |
+| taskExpireTime | 3m0s | task expire time is the time that a task is treated expired if the task is not accessed within the time |
+| peerGCDelay | 3m0s | peer gc delay is the delay time to execute the GC after the peer has reported the offline |
+| gcDiskInterval | 15s | GCDiskInterval is the interval time to execute GC disk |
+| youngGCThreshold | 100GB | if the available disk space is more than YoungGCThreshold and there is no need to GC disk |
+| fullGCThreshold | 5GB | if the available disk space is less than FullGCThreshold and the supernode should gc all task files which are not being used |
+| IntervalThreshold | 2h0m0s | IntervalThreshold is the threshold of the interval at which the task file is accessed |
+| cleanratio | 1 | CleanRatio is the ratio to clean the disk and it is based on 10 |
```
Comment (Starnop) [inline docs/config/supernode_properties.md]: Actually, the `cleanratio` can not be parsed normally. Could you please update it BTY?

File: docs/config/supernode_properties.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+### Some common configurations
+
+Supernode will read the config file in `/etc/dragonfly/supernode.yml` in default.
+In Dragonfly,supernode provides `listenPort` for dfgets to connect, and dfget downloads document from `downloadPort` instead of `listenPort`.
```
Comment (Starnop) [inline docs/config/supernode_properties.md]: Add a space after `In Dragonfly,`

File: docs/config/supernode_properties.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+We can also configure the supernode's IP via `advertiseIP`.
+
+### About gc parameters
+
+In supernode ,gc will begin `gcInitialDelay` time after supernode works.Then supernode will run peer-gc goroutine and task-gc goroutine every  `gcMetaInterval` time.
```
Comment (Starnop) [inline docs/config/supernode_properties.md]: Note the location of the comma.

File: docs/config/supernode_properties.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: `        // LinkLimit is set for supernode to limit every piece download network speed.
        // default: 20 MB, in format of G(B)/g/M(B)/m/K(B)/k/B, pure number will also be parsed as Byte.
        LinkLimit rate.Rate `yaml:"linkLimit"`

        // SystemReservedBandwidth is the network bandwidth reserved for system software.
        // default: 20 MB, in format of G(B)/g/M(B)/m/K(B)/k/B, pure number will also be parsed as Byte.
        SystemReservedBandwidth rate.Rate `yaml:"systemReservedBandwidth"`

        // MaxBandwidth is the network bandwidth that supernode can use.
        // default: 200 MB, in format of G(B)/g/M(B)/m/K(B)/k/B, pure number will also be parsed as Byte.
        MaxBandwidth rate.Rate `yaml:"maxBandwidth"`
`

[the remark this answers is not in the document]
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: > In fact, there is an error here.
> 
> ```
> LinkLimit:               DefaultLinkLimit * rate.MB,
> SystemReservedBandwidth: DefaultSystemReservedBandwidth * rate.MB,
> MaxBandwidth:            DefaultMaxBandwidth * rate.MB,
> ```
> 
> And the default values are as follows:
> 
> ```
> DefaultLinkLimit = 20 * rate.MB
> DefaultSystemReservedBandwidth = 20 * rate.MB
> DefaultMaxBandwidth = 200 * rate.MB
> ```
> 
> They are both multiply by rate.MB. So could you please fix that? @yunfeiyanggzq THX

I noticed it ,I use my computer's default value,I wil fix it

File: docs/config/supernode_properties.md (hunk 4 above)
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: done

File: docs/config/supernode_properties.md (hunk 6 above)
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: done

File: docs/config/supernode_properties.md (hunk 7 above)
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: done

File: docs/config/supernode_properties.md (hunk 5 above)
Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: > Actually, the `cleanratio` can not be parsed normally. Could you please update it BTY?

so do you mean add  the yaml in config.go as follows
`// CleanRatio is the ratio to clean the disk and it is based on 10.
	// It means the value of CleanRatio should be [1-10].
	//
	// default: 1
	CleanRatio int
`
or delete this parameter in this doc

Comment (Starnop) [reply docs/config/supernode_properties.md]: IMOP, delete it for now .

Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: ok

Comment (yunfeiyanggzq) [reply docs/config/supernode_properties.md]: Done

File: docs/config/supernode_properties.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Starnop) [reply docs/config/supernode_properties.md]: THX. And #994 has done that. 😄

File: docs/config/supernode_properties.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  gcMetaInterval: 2m0s
+  taskExpireTime: 2m0s
+  peerGCDelay: 3m0s
+  gcDiskInterval: 15s
+```
+
+When deploying with Docker, you can mount the default path when starting up supernode image with `-v`.
+
+```sh
+docker run -d --name supernode --restart=always -p 8001:8001 -p 8002:8002 -v /etc/dragonfly:/etc/dragonfly dragonflyoss/supernode:0.4.3
```
Comment (Starnop) [inline docs/config/supernode_properties.md]: I think only mount the config file `/etc/dragonfly/supernode.yml` is enough. WDYT?

**Comment by yunfeiyanggzq** [comment]
@Starnop @yeya24  I have added  doc about how to configure dfget and dfdaemon,please review it ,thanks!

File: docs/config/dfdaemon_properties.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+## Parameter instructions
+
+The following startup parameters are supported for `dfdaemon`
+
+| Parameter | Description |
+| ------------- | ------------- |
+| dfget_flags |	dfget properties |
+| dfpath | dfget path |
+| hijack_https | HijackHTTPS is the list of hosts whose https requests should be hijacked by dfdaemon. The first matched rule will be used |
+| localrepo | Temp output dir of dfdaemon, by default `.small-dragonfly/dfdaemon/data/` |
```
Comment (Starnop) [inline docs/config/dfdaemon_properties.md]: default value is `$HOME/.small-dragonfly/dfdaemon/data/`

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**